### PR TITLE
Issue 143

### PR DESCRIPTION
**Pull request by fsanaulla**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 machine:
   timezone: UTC
   java:
-    version: oraclejdk8
+    version: openjdk8
```
Comment (fcrimins) [inline circle.yml]: Why this change from oraclejdk8 to openjdk8?

Comment (fsanaulla) [reply circle.yml]: Was testing env for tests, forgot to change it back.

File: src/main/scala/com/hamstoo/daos/MongoRepresentationDao.scala [hunk 2]
```diff
@@ -90,6 +90,13 @@ class MongoRepresentationDao(db: Future[DefaultDB]) {
     seq <- c.find(d :~ ID -> id).coll[Representation, Seq]()
   } yield seq
 
+  /** Retrieves a current (latest) public representation by URL. */
+  @deprecated("Must be removed in future releases", "1.0.0")
+  def retrieveByUrl(url: String): Future[Option[Representation]] = for {
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/MongoRepresentationDao.scala:128]: What's the purpose of this method?  It appears to be returning any representation, public or private, not only public as it says.

File: src/main/scala/com/hamstoo/models/Comment.scala [hunk 3]
```diff
@@ -32,8 +32,10 @@ case class Comment(
                     pageCoord: Option[PageCoord] = None,
                     memeId: Option[String] = None,
                     timeFrom: Long = DateTime.now.getMillis,
-                    timeThru: Long = Long.MaxValue) {
+                    timeThru: Long = Long.MaxValue) extends Sortable with HasShortcut[CommentShortcut] {
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/models/Comment.scala]: Is it really necessary to `extends Sortable` here?  This seems very Java'y.  See `PageCoord.sortWith` which replaced the same `extends` for `PageCoord`.

File: src/main/scala/com/hamstoo/models/HasShortcut.scala [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.hamstoo.models
+
+/**
+  * Trait that implement shortcut functionality
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/models/HasShortcut.scala]: Please explain what a "shortcut" is.

File: src/main/scala/com/hamstoo/models/Sortable.scala [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.hamstoo.models
+
+/**
+  * Trait that define sort by page coordinates functionality
+  */
+trait Sortable {
+  val pageCoord: Option[PageCoord]
+}
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/models/Sortable.scala]: Let's try to avoid using inheritance to achieve this functionality.  We can use functional `sortWith` instead with duck typing: `.sortWith { case (duck0: {val pageCoord: Option[PageCoord]}, duck1: {val pageCoord: Option[PageCoord]}) => PageCoord.sortWith(duck0.pageCoord, duck1.pageCoord) }`.

File: src/main/scala/com/hamstoo/utils.scala [hunk 6]
```diff
@@ -106,7 +106,7 @@ package object utils {
     * want to display excerpts of them on the My Marks page in human-readable form, (2) we may want to use them
     * as input to our own fit of word2vec or GloVe, and (3) the Conceptnet Numberbatch model is fitted with them
     * included.  This last point is part of what leads to a 418,000-word English vocabulary in which the following
-    * words are all found independently: "can't", "can't", and "can`t".
+    * words are all found independently: "can't", "can't", and "can't".
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/utils.scala:110]: Is some sort of an automatic code formatter happening here?  Why are all the back quotes being changed to forward quotes?  This also happened in MongoRepresentationDao.scala.
